full review pre-trial 03 (#19)
epiverse-trace/tutorials-late: episodes/simulating-transmission.Rmd

```{r traj, echo = FALSE, message = FALSE, fig.width = 10, eval = TRUE}
# load contact and population data from socialmixr::polymod
polymod <- socialmixr::polymod
contact_data <- socialmixr::contact_matrix(
  polymod,
  countries = "United Kingdom",
  age.limits = c(0, 20, 40),
  symmetric = TRUE
)

# prepare contact matrix
contact_matrix <- t(contact_data$matrix)

# prepare the demography vector
demography_vector <- contact_data$demography$population
names(demography_vector) <- rownames(contact_matrix)

# initial conditions: one in every 1 million is infected
initial_i <- 1e-6
initial_conditions_inf <- c(
  S = 1 - initial_i, E = 0, I = initial_i, R = 0, V = 0
)

initial_conditions_free <- c(
  S = 1, E = 0, I = 0, R = 0, V = 0
)

# build for all age groups
initial_conditions <- rbind(
  initial_conditions_inf,
  initial_conditions_free,
  initial_conditions_free
)
rownames(initial_conditions) <- rownames(contact_matrix)

# prepare the population to model as affected by the epidemic

> uk_population <- population(

Author: appropriate edit suggestion!

)
```

For the age categories that are free from infection, the initial conditions are $S(0)=1$, $E(0) =0$, $I(0)=0$, $R(0)=0$. We specify this as follows,

```{r initial_free}
initial_conditions_free <- c(
  S = 1, E = 0, I = 0, R = 0, V = 0
)
```

We combine the three initial conditions vectors into one matrix, 

```{r initial condtions}
# combine the initial conditions
initial_conditions <- rbind(
  initial_conditions_inf, # age group 1
  initial_conditions_free, # age group 2
  initial_conditions_free # age group 3
)

# use contact matrix to assign age group names
rownames(initial_conditions) <- rownames(contact_matrix)
initial_conditions
```

### 3. Population structure
The population object requires a vector containing the demographic structure of the population. The demographic vector must be a named vector containing the number of individuals in each age group of our given population. In this example, we can extract the demographic information from the `contact_data` object that we obtained using the `socialmixr` package.

```{r demography}
demography_vector <- contact_data$demography$population
names(demography_vector) <- rownames(contact_matrix)
demography_vector
```

> To create our population object, we call the function `population()` specifying a name, the contact matrix, the demography vector and the initial conditions.
> 
> ```{r population}
> uk_population <- population(
>   name = "UK",
>   contact_matrix = contact_matrix,
>   demography_vector = demography_vector,
>   initial_conditions = initial_conditions
> )
> ```

Author: ```suggestion
To create our population object, from the {epidemics} package we call the function `population()` specifying a name, the contact matrix, the demography vector and the initial conditions.

```{r population}
uk_population <- epidemics::population(
  name = "UK",
  contact_matrix = contact_matrix,
  demography_vector = demography_vector,
  initial_conditions = initial_conditions
)
```
```

### 4. Model parameters

To run our model we need to specify the model parameters: 

- transmission rate $\beta$,
- infectiousness rate $\alpha$ (preinfectious period=$1/\alpha$),
- recovery rate $\gamma$ (infectious period=$1/\gamma$).

In `epidemics`, we specify the model inputs as :

- `transmission_rate` = $R_0 \gamma$,
- `infectiousness_rate` = $\alpha$,
- `recovery_rate` = $\gamma$, 
  

> We will simulate a strain of influenza with pandemic potential with $R_0=1.46$, a preinfectious period of 3 days and infectious period of 7 days. Therefore our inputs will be:

Author: Which one is clearer? `latent` period or `preinfectious` period?

```suggestion
We will simulate a strain of influenza with pandemic potential with $R_0=1.46$, a preinfectious period of 3 days and infectious period of 7 days.

- `latent_period <- 3.0`
- `infectious_period <- 7.0`
- `basic_reproduction <- 1.46`

Therefore our inputs will be:
```

> - `transmission_rate = 1.46 / 7.0`,
> - `infectiousness_rate = 1.0 / 3.0`,
> - `recovery_rate = 1.0 / 7.0`.

Author: For consisteny with the callout box bellow

```suggestion
- `infectiousness_rate = 1/latent_period`,
- `recovery_rate = 1/infectious_period`,
- `transmission_rate =  basic_reproduction*recovery_rate`.
```

::::::::::::::::::::::::::::::::::::::::::::::::

Now we are ready to run our model. Let's load the `epidemics` package : 

```{r loading}
library(epidemics)
```

Then we specify `time_end=600` to run the model for 600 days.
```{r run_model}
output <- model_default(
  population = uk_population,
  transmission_rate = 1.46 / 7.0,
  infectiousness_rate = 1.0 / 3.0,
  recovery_rate = 1.0 / 7.0,
  time_end = 600, increment = 1.0
)
head(output)
```

*Note : This model also has the functionality to include vaccination and tracks the number of vaccinated individuals through time. Even though we have not specified any vaccination, there is still a vaccinated compartment in the output (containing no individuals). We will cover the use of vaccination in future tutorials.*

Our model output consists of the number of individuals in each compartment in each age group through time. We can visualise the infectious individuals only (those in the $I$ class) through time.

```{r visualise, fig.width = 10}

> filter(output_plot, compartment %in% c("exposed", "infectious")) %>%

Author: ```suggestion
library(tidyverse)

filter(output, compartment %in% c("exposed", "infectious")) %>%
```
